cou/steps/analyze.py

```diff
@@ -240,41 +196,19 @@ def _get_minimum_cloud_series(self) -> Optional[str]:
         :rtype: Optional[str]
         """
         return min(
-            (app.series for app in self.apps_control_plane + self.apps_data_plane),
+            (app.series for app in self.apps),
             default=None,
         )
 
     @property
-    def data_plane_machines(self) -> dict[str, juju_utils.Machine]:
-        """Data-plane machines of the model.
-
-        :return: Data-plane machines of the model.
-        :rtype: dict[str, Machine]
-        """
-        return {
-            machine_id: app.machines[machine_id]
-            for app in self.apps_data_plane
-            for machine_id in app.machines
-        }
-
-    @property
-    def control_plane_machines(self) -> dict[str, juju_utils.Machine]:
-        """Control-plane machines of the model.
+    def machines(self) -> dict[str, juju_utils.Machine]:
+        """All OpenStack machines of the model.
 
-        :return: Control-plane machines of the model.
+        :return: All OpenStack machines of the model.
         :rtype: dict[str, Machine]
         """
         return {
             machine_id: app.machines[machine_id]
-            for app in self.apps_control_plane
+            for app in self.apps
             for machine_id in app.machines
         }
```

Comment:
This is basically what juju_utils already does:

https://github.com/canonical/charmed-openstack-upgrader/blob/5bfa932bb459377c7311373d8a3284442a7afc20/cou/utils/juju_utils.py#L279-L294

I prefer passing a new class attribute  `machines` to the Analysis object instead of having kind of duplicate code

Reply:
Ah so this was kind of a refactoring - there was already this method and this logic, along with other methods to get control plane machines and data plane machines.  They weren't really being used, so I factored them out and simplified this method.  Maybe I should revert the refactoring and do it in another PR. Thanks for flagging the duplication; even more that can be factored out. :)

Reply:
I've removed this refactoring